api/v1/iprangeclaim_types.go

```diff
@@ -0,0 +1,144 @@
+
+	IpAddresses []string `json:"ipAddresses,omitempty"`
+
+	IpAddressesDotDecimal []string `json:"ipAddressesDotDecimal,omitempty"`
+
+	StartAddress string `json:"startAddress,omitempty"`
+
+	StartAddressDotDecimal string `json:"startAddressDotDecimal,omitempty"`
+
+	EndAddress string `json:"endAddress,omitempty"`
+
+	EndAddressDotDecimal string `json:"endAddressDotDecimal,omitempty"`
+
+	IpRangeName string `json:"ipAddressName,omitempty"`
+
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+//+kubebuilder:printcolumn:name="IpRange",type=string,JSONPath=`.status.ipRange`
+//+kubebuilder:printcolumn:name="IpRangeAssigned",type=string,JSONPath=`.status.conditions[?(@.type=="IpRangeAssigned")].status`
+//+kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].status`
+//+kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +kubebuilder:resource:shortName=irc
+
+// IpRangeClaim is the Schema for the iprangeclaims API
+type IpRangeClaim struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   IpRangeClaimSpec   `json:"spec,omitempty"`
+	Status IpRangeClaimStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// IpRangeClaimList contains a list of IpRangeClaim
+type IpRangeClaimList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IpRangeClaim `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&IpRangeClaim{}, &IpRangeClaimList{})
+}
+
+var ConditionIpRangeClaimReadyTrue = metav1.Condition{
+	Type:    "Ready",
+	Status:  "True",
+	Reason:  "IpRangeResourceReady",
+	Message: "Ip Range Resource is ready",
+}
+
+var ConditionIpRangeClaimReadyFalse = metav1.Condition{
+	Type:    "Ready",
+	Status:  "False",
+	Reason:  "IpRangeResourceNotReady",
+	Message: "Ip Range Resource is not ready",
+}
+
+var ConditionIpRangeClaimReadyFalseStatusGen = metav1.Condition{
+	Type:    "Ready",
+	Status:  "False",
+	Reason:  "IpRangeClaimStatusGenerationFailed",
+	Message: "Failed to generate Ip Range Status",
+}
+
+var ConditionIpRangeAssignedTrue = metav1.Condition{
+	Type:    "IpRangeAssigned",
+	Status:  "True",
+	Reason:  "IpRangeCRCreated",
+	Message: "New Ip Range fetched from NetBox and IpRange CR was created",
+}
+
+var ConditionIpRangeAssignedFalse = metav1.Condition{
+	Type:    "IpRangeAssigned",
+	Status:  "False",
+	Reason:  "IpRangeCRNotCreated",
+	Message: "Failed to fetch new Ip Range from NetBox",
+}
+
+var ConditionIpRangeAssignedFalseSizeMissmatch = metav1.Condition{
+	Type:    "IpRangeAssigned",
+	Status:  "False",
+	Reason:  "IpRangeCRNotCreated",
+	Message: "Assigned/Resored ip range has less available ip addresses than requested",
+}
```

Comment:
I think we use IpAddress at other places already, so maybe `Ip` would be more consistent. :)

Reply:
I changed it to 'IP' in the Type, Status, Reason and Message fields, so it's consistent with the conditions of the other types. In the other parts of the code we use 'Ip' because it's easier to read.